src/optimizer/query_rewrite.c

```diff
@@ -1844,6 +1851,63 @@ qo_reduce_equality_terms (PARSER_CONTEXT * parser, PT_NODE * node, PT_NODE ** wh
       *orgp = parser_append_node (join_term_list, *orgp);
     }
 
+  /* remove always-true term */
+  while ((expr = ((expr_prev) ? expr_prev->next : *orgp)))
+    {
+      PT_OP_TYPE op = expr->info.expr.op;
+      cut_off = false;
+      opd1 = expr->info.expr.arg1;
+      opd2 = expr->info.expr.arg2;
+
+      if (expr->or_next == NULL)
+	{
+	  if (opd1 && opd2 && op == PT_EQ && opd1->node_type == PT_VALUE && opd2->node_type == PT_VALUE)
+	    {
+	      domain = pt_xasl_node_to_domain (parser, expr);
+	      qualifier = expr->info.expr.qualifier;
+	      dbv1 = pt_value_to_db (parser, opd1);
+	      dbv2 = pt_value_to_db (parser, opd2);
+	      opd3 = expr->info.expr.arg3;
+	      dbv3 = pt_value_to_db (parser, opd3);
+	      if (pt_evaluate_db_value_expr
+		  (parser, expr, expr->info.expr.op, dbv1, dbv2, dbv3, &dbval_res, domain, opd1, opd2, opd3, qualifier))
+		{
+		  if (DB_VALUE_TYPE (&dbval_res) == DB_TYPE_INTEGER && db_get_int (&dbval_res) == 1)
+		    {
+		      cut_off = true;
+		    }
+		}
+	    }
+	}
+      else
+	{
+	  /*
+	   * give up
+	   */
+	  ;
+	}
+
+      if (cut_off)
+	{
+	  /* cut if off from CNF list */
+	  if (expr_prev)
+	    {
+	      expr_prev->next = expr->next;
+	    }
+	  else
+	    {
+	      *orgp = expr->next;
+	    }
+	  expr->next = NULL;
+	  parser_free_tree (parser, expr);
+	}
+      else
+	{
+	  expr_prev = (expr_prev) ? expr_prev->next : expr;
```

Comment:
Why should we check expr_prev?
What problems will occur if we do the following?
```suggestion
	  expr_prev = expr;
```

Reply:
The initial value of expr_prev is set to NULL. In this scenario, expr is pointing to the first element in the list. It seems likely that an error will occur when first entering the while loop. This implementation method was referenced from the where_type function.